mmv1/products/dataplex/Datascan.yaml

```diff
@@ -0,0 +1,640 @@
+                 Rows with null values will automatically fail a rule, unless ignoreNull is true. In that case, such null rows are trivially considered passing. Only applicable to ColumnMap rules.
+             - !ruby/object:Api::Type::String
+               name: 'dimension'
+               required: true
+               description: |
+                 The dimension a rule belongs to. Results are also aggregated at the dimension level. Supported dimensions are ["COMPLETENESS", "ACCURACY", "CONSISTENCY", "VALIDITY", "UNIQUENESS", "INTEGRITY"]
+             - !ruby/object:Api::Type::Integer
+               name: 'threshold'
+               description: |
+                The minimum ratio of passing_rows / total_rows required to pass this rule, with a range of [0.0, 1.0]. 0 indicates default value (i.e. 1.0).
+               default_value: 0
+             - !ruby/object:Api::Type::NestedObject
+               name: 'rangeExpectation'
+               description: |
+                 ColumnMap rule which evaluates whether each column value lies between a specified range.
+               properties:
+                 - !ruby/object:Api::Type::String
+                   name: 'minValue'
+                   description: |
+                     The minimum column value allowed for a row to pass this validation. At least one of minValue and maxValue need to be provided.
+                 - !ruby/object:Api::Type::String
+                   name: maxValue
+                   description: |
+                     The maximum column value allowed for a row to pass this validation. At least one of minValue and maxValue need to be provided.
+                 - !ruby/object:Api::Type::Boolean
+                   name: 'strictMinEnabled'
+                   default_value: false
+                   description: |
+                     Whether each value needs to be strictly greater than ('>') the minimum, or if equality is allowed. 
+                     Only relevant if a minValue has been defined. Default = false.
+                 - !ruby/object:Api::Type::Boolean
+                   name: 'strictMaxEnabled'
+                   default_value: false
+                   description: |
+                     Whether each value needs to be strictly lesser than ('<') the maximum, or if equality is allowed.
+                     Only relevant if a maxValue has been defined. Default = false.
+             - !ruby/object:Api::Type::NestedObject
+               name: 'nonNullExpectation'
+               description: |
+                 ColumnMap rule which evaluates whether each column value is null.
+             - !ruby/object:Api::Type::NestedObject
+               name: 'setExpectation'
+               description: |
+                 ColumnMap rule which evaluates whether each column value is contained by a specified set.
+               properties:
+                 - !ruby/object:Api::Type::Array
+                   name: 'values'
```

Comment:
i think we'll want to make this `required: true`

Reply:
I'd want to defer adding a required:true here in case we add a later field here which will have one-of semantics with the 'values' field. For example, if we allow the user to specify a SQL substatement to fetch the specified set instead of a static list.


I'm assuming that marking a field required cannot be undone later, but it if can, marking it required for now should be okay.

Reply:
i believe marking it `required` now and changing to `optional` later should _not_ be a breaking change. i think going the other way (`optional` to `required`) is breaking, but if a user is required to have it in their config, then later changes to optional, there is not change to their config that they'd need to make. does that make sense?

alternatively, if we had it as `optional` then changed it to `required`, a user who hadn't defined the field previously, would then need to start defining it - which would be a breaking change.